rascal2/widgets/sliders_view.py

```diff
@@ -0,0 +1,550 @@
+        self.update_slider_display_from_property(in_constructor)
+
+    def update_slider_display_from_property(self, in_constructor: bool) -> None:
+        """Change internal sliders parameters and their representation in GUI
+        if property, underlying sliders parameters have changed.
+
+        Bound to event received from delegate when table values are changed.
+        """
+        # note the order of methods in comparison. Should be as here, as may break
+        # property updates in constructor otherwise.
+        if not (self._updated_from_rascal_property() or in_constructor):
+            return
+
+        self._value_range = self._value_max - self._value_min
+        # the change in property value per single step slider move
+        self._value_step = self._value_range / self._slider_max_idx
+
+        if in_constructor:
+            return
+        # otherwise, update slider's labels
+        self.set_slider_gui_position(self._value)
+        tick_step = self._value_range / self._num_slider_ticks
+        for idx in range(0, self._num_slider_ticks + 1):
+            tick_value = self._value_min + idx * tick_step
+            self._labels[idx].setText(self._tick_label_format.format(tick_value))
+
+    def _updated_from_rascal_property(self) -> bool:
+        """Check if rascal property values related to slider widget have changed.
+        Update them if they are and return True. False if they have not been changed.
+        """
+        updated = False
+        if self._value_min != self._prop.param.min:
+            self._value_min = self._prop.param.min
+            updated = True
+        if self._value_max != self._prop.param.max:
+            self._value_max = self._prop.param.max
+            updated = True
+        if self._value != self._prop.param.value:
+            self._value = self._prop.param.value
+            updated = True
+        return updated
+
+    def _value_to_slider_pos(self, value: float) -> int:
+        """Convert double value into slider position"""
+        return int(round(self._slider_max_idx * (value - self._value_min) / self._value_range, 0))
+
+    def _slider_pos_to_value(self, index: int) -> float:
+        """Convert slider GUI position (index) into double value"""
+        value = self._value_min + index * self._value_step
+        if value > self._value_max:  # This should not happen but do occur due to round-off errors
+            value = self._value_max
+        return value
+
+    def _build_slider(self, initial_value: float) -> QtWidgets.QSlider:
+        """Construct slider widget with integer scales and ticks in integer positions
+        Part of slider constructor
+        """
+
+        slider = QtWidgets.QSlider(QtCore.Qt.Orientation.Horizontal)
+        slider.setMinimum(0)
+        slider.setMaximum(self._slider_max_idx)
+        slider.setTickInterval(self._ticks_step)
+        slider.setSingleStep(self._slider_max_idx)
```

Comment:
please add `slider.setTracking(False)` for now due to performance when dragging we can optimize an remove later